bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedHandler.java

```diff
@@ -0,0 +1,439 @@
+                break;
+            case CHANNEL_SECONDARY_WHITE:
+                if (command instanceof PercentType) {
+                    sendGetRequest("/win&W2=" + new BigDecimal(command.toString()).multiply(new BigDecimal(2.55)));
+                }
+                break;
+            case CHANNEL_MASTER_CONTROLS:
+                if (command instanceof OnOffType) {
+                    if (OnOffType.OFF.equals(command)) {
+                        sendGetRequest("/win&TT=250&T=0");
+                    } else {
+                        sendGetRequest("/win&TT=1000&T=1");
+                    }
+                } else if (command instanceof IncreaseDecreaseType) {
+                    if (IncreaseDecreaseType.INCREASE.equals(command)) {
+                        if (masterBrightness.intValue() < 240) {
+                            sendGetRequest("/win&TT=1000&A=~15"); // 255 divided by 15 = 17 different levels
+                        } else {
+                            sendGetRequest("/win&TT=1000&A=255");
+                        }
+                    } else {
+                        if (masterBrightness.intValue() > 15) {
+                            sendGetRequest("/win&TT=1000&A=~-15");
+                        } else {
+                            sendGetRequest("/win&TT=1000&A=0");
+                        }
+                    }
+                } else if (command instanceof HSBType) {
+                    if ((((HSBType) command).getBrightness()) == PercentType.ZERO) {
+                        sendGetRequest("/win&TT=500&T=0");
+                    }
+                    masterBrightness = new BigDecimal((((HSBType) command).getBrightness()).toString())
+                            .multiply(new BigDecimal(2.55));
+                    primaryColor = new HSBType(command.toString());
+                    if (primaryColor.getSaturation().intValue() < config.saturationThreshold) {
+                        sendWhite();
+                    } else if (primaryColor.getSaturation().intValue() == 32 && primaryColor.getHue().intValue() == 36
+                            && hasWhite) {
+                        // Google sends this when it wants white
+                        sendWhite();
+                    } else {
+                        sendGetRequest(
+                                "/win&TT=1000&FX=0&CY=0&CL=" + createColorHex(primaryColor) + "&A=" + masterBrightness);
+                    }
+                } else {// should only be PercentType left
+                    masterBrightness = new BigDecimal(command.toString()).multiply(new BigDecimal(2.55));
+                    sendGetRequest("/win&TT=1000&A=" + masterBrightness);
+                }
+                return;
+            case CHANNEL_PRIMARY_COLOR:
+                if (command instanceof OnOffType) {
+                    return;
+                } else if (command instanceof HSBType) {
+                    primaryColor = new HSBType(command.toString());
```

Comment:
Why are you creating a new HSBType when you can cast the command?
```suggestion
                    primaryColor = (HSBType) command;
```